### PR TITLE
fix: filter out instead of remove

### PR DESCRIPTION
**Pull request by Julieta11**
We are not correctly filtering out* events and this is a problem because we are sending empty messages from events like delete or with empty state keys as part of the message history.

*Filter out all org.decentraland.friendship events where:
- The body is empty OR
- The state key is empty OR
- The event is not a request

### CODE REVIEW

File: src/ConversationCursor.ts [hunk 1]
```diff
@@ -29,7 +34,18 @@ export class ConversationCursor {
     getMessages(): TextMessage[] {
         const latestReadTimestamp: Timestamp | undefined = this.lastReadMessageTimestampFetch(this.roomId)?.timestamp
 
-        const events = this.window.getEvents()
+        // We filter out all friendship events to only keep those that are requests, have a message body
+        // and have a state key, which indicates that the event was sent by the requester.
+        const events = this.window.getEvents().filter(event => {
```
Comment (agusaldasoro) [inline src/ConversationCursor.ts:39]: should this be moved to a method in the `src/Utils.ts` file?

File: src/Utils.ts [hunk 2]
```diff
@@ -210,3 +210,19 @@ const GET_MESSAGES_AND_FRIENDSHIP_EVENTS_FILTER = (userId: SocialId, limit?: num
...
     })
+
+/**
+ * We filter out all friendship events that are not requests, does not have a message body
+ * or does not have a state key.
+ * @param events
+ * @returns the length of the filtered array
+ * @internal
+ */
+export function eventsToFilterOut(events: MatrixEvent[]): number {
```
Comment (agusaldasoro) [inline src/Utils.ts:221]: can this method re-use the code above in `ConversationCursor`?

**Comment by Julieta11** [comment]
Note: I am modifying the filter logic from what it was before because I previously tried to get the undesired set to remove them from the timeline. Now, I am filtering the events that meet the required condition to be considered as a part of the message history.

File: src/ConversationCursor.ts [hunk 3]
```diff
@@ -100,29 +91,23 @@ export class ConversationCursor {
...
 
             gotResults = true
             while (windowSize < initialSize && gotResults) {
                 gotResults = await window.paginate(EventTimeline.FORWARDS, initialSize - windowSize)
 
-                eventsToFilter = eventsToFilterOut(window.getEvents())
-
-                windowSize = window.getEvents().length - eventsToFilter
+                windowSize = filterEvents(window.getEvents()).length
             }
```
Comment (agusaldasoro) [inline src/ConversationCursor.ts:111]: is it okay to return a `windowsSize` bigger than `initialSize`?

**Review by agusaldasoro — APPROVED** [review]
Great work!

**Comment by Julieta11** [comment]
Closing this PR as the approach to support sending the request event message as part of the chat history won't be implemented by filtering the `timeline` or the `room.timeline`. At the moment of filtering, the timeline does not contain all events, making it impossible to remove them all.

**Pull request closed without merging**